### ISSUE TITLE
Highway networks - freezing?

### ISSUE THREAD

**Issue by HristoBuyukliev** [fchollet/keras#1601]
Hi,

I have a problem when creating highway networks of 20+ layers. The scripts hangs while initializing the net, giving no error nor warning. Note that it can create 4 layer net with 1000 hidden units (so 4m connections), and hangs when creating 20 layers with 40 hidden units (so 1600*20 = 32k connections).

Here is the code in question:

	X_train, y_train, X_test, y_test = hacking_script.load_data()
	n_layers = 100
	dropout = 0.1
	model = Sequential()
	model.add(layers.core.Dropout(dropout, input_shape=(38,)))
	for layer in range(n_layers):
		model.add(layers.core.Highway(activation='relu'))
		model.add(layers.core.Dropout(dropout))
	model.add(layers.core.Dense(1, activation='sigmoid'))

	model.compile(loss='binary_crossentropy', optimizer='adagrad')
	early_stopping = EarlyStopping(monitor='val_loss', patience=3)
	model.fit(X_train, y_train, batch_size=100, nb_epoch=50, 
		show_accuracy=True, validation_data=(X_test, y_test), callbacks=[early_stopping])
	predictions = model.predict_proba(X_test)
	print(metrics.roc_auc_score(y_test, predictions))
